CPAC/registration/registration.py

```diff
@@ -1820,6 +1827,197 @@ def register_ANTs_anat_to_template(wf, cfg, strat_pool, pipe_num, opt=None):
+            merge_xfms, 'in_files')
+
+        # applywarp --rel --interp=spline -i ${T1wRestore} -r ${Reference} -w ${OutputTransform} -o ${OutputT1wImageRestore}
+        fsl_apply_warp_t1_to_template = pe.Node(interface=fsl.ApplyWarp(),
+                                                name=f'FSL-ABCD_T1_to_template_{pipe_num}')
+        fsl_apply_warp_t1_to_template.inputs.relwarp = True
+        fsl_apply_warp_t1_to_template.inputs.interp = 'spline'
+
+        node, out = strat_pool.get_data('desc-restore_T1w')
+        wf.connect(node, out, fsl_apply_warp_t1_to_template, 'in_file')
+
+        node, out = strat_pool.get_data('T1w_template')
+        wf.connect(node, out, fsl_apply_warp_t1_to_template, 'ref_file')
+
+        wf.connect(merge_xfms, 'merged_file', 
+            fsl_apply_warp_t1_to_template, 'field_file')
+
+        # applywarp --rel --interp=nn -i ${T1wRestoreBrain} -r ${Reference} -w ${OutputTransform} -o ${OutputT1wImageRestoreBrain}
+        fsl_apply_warp_t1_brain_to_template = pe.Node(interface=fsl.ApplyWarp(),
+                                                    name=f'FSL-ABCD_T1_brain_to_template_{pipe_num}')
+        fsl_apply_warp_t1_brain_to_template.inputs.relwarp = True
+        fsl_apply_warp_t1_brain_to_template.inputs.interp = 'nn'
+
+        # TODO connect T1wRestoreBrain, check T1wRestoreBrain quality
+        node, out = strat_pool.get_data('desc-brain_T1w')
+        wf.connect(node, out, fsl_apply_warp_t1_brain_to_template, 'in_file')
+
+        node, out = strat_pool.get_data('T1w_template')
+        wf.connect(node, out, fsl_apply_warp_t1_brain_to_template, 'ref_file')
+
+        wf.connect(merge_xfms, 'merged_file', 
+            fsl_apply_warp_t1_brain_to_template, 'field_file')
+
+        fsl_apply_warp_t1_brain_mask_to_template = pe.Node(interface=fsl.ApplyWarp(),
+                                                        name=f'FSL-ABCD_T1_brain_mask_to_template_{pipe_num}')
+        fsl_apply_warp_t1_brain_mask_to_template.inputs.relwarp = True
+        fsl_apply_warp_t1_brain_mask_to_template.inputs.interp = 'nn'
+
+        node, out = strat_pool.get_data('space-T1w_desc-brain_mask')
+        wf.connect(node, out, fsl_apply_warp_t1_brain_mask_to_template, 'in_file')
+
+        node, out = strat_pool.get_data('T1w_template')
+        wf.connect(node, out, fsl_apply_warp_t1_brain_mask_to_template, 'ref_file')
+
+        wf.connect(merge_xfms, 'merged_file', 
+            fsl_apply_warp_t1_brain_mask_to_template, 'field_file')
+
+        # fslmaths ${OutputT1wImageRestore} -mas ${OutputT1wImageRestoreBrain} ${OutputT1wImageRestoreBrain}
+        apply_mask = pe.Node(interface=fsl.maths.ApplyMask(),
+                            name=f'get_t1_brain_{pipe_num}')
+
+        wf.connect(fsl_apply_warp_t1_to_template, 'out_file',
+            apply_mask, 'in_file')
+        
+        wf.connect(fsl_apply_warp_t1_brain_to_template, 'out_file',
+            apply_mask, 'mask_file')
+
```

Comment:
@sgiavasis @HechengJin0 Do you think we need to take this node out and put it in a separate function so that Hecheng can use it to apply functional mask in standard space in NHP pipeline?

Reply:
Can this node block operate without `desc-restore` present, if we want to run a simpler pipeline and warp the T1 to standard?

To address your question: does the warping of the mask to template rely on anything else in the node block? If not, it may be better to make it its own node block - not only for re-use, but also to avoid making the provenance tracking look like the step of warping the brain mask to template depended on warping the actual T1 to template. I'm including this consideration into the 1.8+ dev guide notes also.

Reply:
Hecheng will make changes in NHP branch